elastic/kibana
# [Stateful sidenav] Feedback button

In this PR I've added a feedback btn in the new solution side nav **for non trial users**.

As long as a deployment is in the trial dates, the button does not show. When the trial ends, if the deployment is still running, then the button appears.

The button can be dismissed, but the state is in `sessionStorage`. That means that it reappears when Kibana is opened in a new browser window or tab.

## How to test

* Add the following in your `kibana.dev.yml`

```yml
xpack.cloud.id: "ftr_fake_cloud_id:aGVsbG8uY29tOjQ0MyRFUzEyM2FiYyRrYm4xMjNhYmM="
xpack.cloud.base_url: "https://cloud.elastic.co"
```
* Launch Kibana and setup a solution on the space
* You should see the feedback button
* Add the following to the `kibana.dev.yml`

```yml
# Any date in the future
xpack.cloud.trial_end_date: "2024-10-20T10:10:00Z"
```
* Refresh the browser window
* The button should not appear

## Screenshots

<table>
  <tr>
    <td valign="top"><img width="264" alt="Screenshot 2024-10-10 at 12 43 54" src="https://github.com/user-attachments/assets/c26a7c79-9a12-4dcc-a3ae-e1bac53342fe"></td>
    <td valign="top"><img width="254" alt="Screenshot 2024-10-10 at 12 44 38" src="https://github.com/user-attachments/assets/dc28f5e7-c8b0-4b38-9950-455f20e56cf8"></td>
    <td valign="top"><img width="267" alt="Screenshot 2024-10-10 at 12 44 54" src="https://github.com/user-attachments/assets/66308b7b-2f92-43dc-97c9-90b0a79b0387"></td>
  </tr>
  <tr>
    <td><img width="259" alt="Screenshot 2024-10-10 at 12 44 08" src="https://github.com/user-attachments/assets/0e809f56-f8df-41b6-8c82-02526684b20f"></td>
    <td></td>
    <td></td>
  </tr>
</table>


Fixes https://github.com/elastic/kibana-team/issues/1135
